packages/flutter/lib/src/painting/text_painter.dart

```diff
@@ -1066,22 +1262,16 @@ class TextPainter {
         continue;
       }
       final TextBox box = boxes.first;
-
-      if (prevCodeUnit == NEWLINE_CODE_UNIT) {
-        return Rect.fromLTRB(_emptyOffset.dx, box.bottom, _emptyOffset.dx, box.bottom + box.bottom - box.top);
-      }
-
-      final double caretEnd = box.end;
-      final double dx = box.direction == TextDirection.rtl ? caretEnd - caretPrototype.width : caretEnd;
-      return Rect.fromLTRB(clampDouble(dx, 0, _paragraph!.width), box.top,
-          clampDouble(dx, 0, _paragraph!.width), box.bottom);
+      return prevCodeUnit == NEWLINE_CODE_UNIT
```

Comment:
This isn't introduced in this PR so I'll leave that as is.

Reply:
Do we have an issue on file for this? Maybe link that in a TODO here?

Reply:
The exponential search itself is a bit problematic, as it could miss newline characters during the search when the search step becomes too huge. SkParagraph already exposed a method for getting the glyph cluster information at a given code unit offset. We'll switch to that soon(TM).